### PR TITLE
Tc1 event selection click

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- Update the UI (event time buttons) when the user adjusts the given event times
- Allow a user to change the date that an event takes place
- Incorporate useContext to keep track of the events the user has selected
- Add on click functionality to select different events to add to google calendar

<what will be done in later PRs and not included here>

- add events to google calendar on click of conform events button

## Milestones
<the milestones or stories/features that this works towards>

- TC1 frontend response to user input

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
<div>
    <a href="https://www.loom.com/share/4319893fac0340c5975fcb22b91275fb">
      <p>Cook throughout the week: User Time Preference and Event Selection Demo - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/4319893fac0340c5975fcb22b91275fb">
      
    </a>
  </div>
<div>
    <a href="https://www.loom.com/share/d778e4a003f84a10ac827749394c83dd">
      <p>Single Prep Day: User Time Preference and Event Selection Demo - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/d778e4a003f84a10ac827749394c83dd">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/components/CalendarEventCard.tsx [hunk 1]
```diff
@@ -8,20 +8,39 @@ import {
...
   const startDate = new Date(eventOption.timeOptions[0].start);
+  const { selectedEvents, setSelectedEvents } = useSelectedEvents();
+
+  const toggleSelection = (timeBlock: GPTimeBlockType) => {
+    const clickedEvent = {
+      ...eventOption,
+      timeOptions: [timeBlock],
+    };
+    // if not already in the set, add
+    const isSelected = selectedEvents.filter((selectedEvent) => selectedEvent.name === clickedEvent.name && selectedEvent.timeOptions[0].start === clickedEvent.timeOptions[0].start && selectedEvent.timeOptions[0].end === clickedEvent.timeOptions[0].end)
```
Comment (paulshaoyuqiao) [inline frontend/src/components/CalendarEventCard.tsx:37]: maybe keep this multiple line so it is easier to read? 
Also can be potentially refactor the equality comparison here as a method directly on the event object? 
this is where declaring things as a class come in handy :)
you can just have `selectedEvent.isEqual(clickedEvent)` while under the hood, you can implement all the `==` logic without frontend having to know

File: frontend/src/components/CalendarEventCard.tsx [hunk 2]
```diff
@@ -8,20 +8,39 @@ import {
...
+    if (isSelected.length === 0) {
+      setSelectedEvents((prev) => [...prev, clickedEvent])
+    } else {
+      setSelectedEvents((prev) => prev.filter((selectedEvent) => selectedEvent.name !== clickedEvent.name && selectedEvent.timeOptions[0].start !== clickedEvent.timeOptions[0].start && selectedEvent.timeOptions[0].end !== clickedEvent.timeOptions[0].end))
```
Comment (paulshaoyuqiao) [inline frontend/src/components/CalendarEventCard.tsx:41]: same here

File: frontend/src/components/CalendarEventCard.tsx [hunk 3]
```diff
@@ -95,19 +114,20 @@ const CalendarEventCard = ({ eventOption, groupNum }: GPCalendarOption) => {
                     minute: "numeric",
                   }
                 );
+                const clickedEvent: GPRecipeEventOptionType = {
+                  ...eventOption,
+                  timeOptions: [option],
+                }
+                const isSelected = selectedEvents.filter((selectedEvent) => selectedEvent.name === clickedEvent.name && selectedEvent.timeOptions[0].start === clickedEvent.timeOptions[0].start && selectedEvent.timeOptions[0].end === clickedEvent.timeOptions[0].end)
```
Comment (paulshaoyuqiao) [inline frontend/src/components/CalendarEventCard.tsx:121]: same here

**Review by paulshaoyuqiao — CHANGES_REQUESTED** [review]
